### PR TITLE
crypto: v2 address format

### PR DESCRIPTION
**Pull request by redshiftzero**
This PR is going to stage the various address-breaking changes we have planned prior to mainnet:

- [x] Uses new poseidon parameters by removing the pinned branch called `oldparams`
- [x] https://github.com/penumbra-zone/penumbra/issues/1077
- [x] https://github.com/penumbra-zone/penumbra/issues/1079
- [x] https://github.com/penumbra-zone/penumbra/issues/1078
- [x] Add v2 address prefix

### CODE REVIEW

**Comment by redshiftzero** [comment]
(smoke test failing due to invalid addresses in allocations file)

**Comment by plaidfinch** [comment]
@redshiftzero Galileo may need to be modified to parse the new address format; I forget exactly which regex is used. We can also yolo and find out though.

File: crypto/src/note.rs [hunk 1]
```diff
@@ -403,11 +403,11 @@ impl From<&Note> for [u8; NOTE_LEN_BYTES] {
...
-        bytes[1..12].copy_from_slice(&note.diversifier.0);
-        bytes[12..20].copy_from_slice(&note.value.amount.to_le_bytes());
-        bytes[20..52].copy_from_slice(&note.value.asset_id.0.to_bytes());
-        bytes[52..84].copy_from_slice(&note.note_blinding.to_bytes());
-        bytes[84..116].copy_from_slice(&note.transmission_key.0);
+        bytes[1..17].copy_from_slice(&note.diversifier.0);
+        bytes[17..25].copy_from_slice(&note.value.amount.to_le_bytes());
+        bytes[25..57].copy_from_slice(&note.value.asset_id.0.to_bytes());
+        bytes[57..89].copy_from_slice(&note.note_blinding.to_bytes());
+        bytes[89..121].copy_from_slice(&note.transmission_key.0);
```
Comment (hdevalence) [inline crypto/src/note.rs:410]: While we're at it, we should remove the `NOTE_TYPE` byte. We don't use it currently, and if we want to version note formats, we should be doing that at the level of the protos, not inside the ciphertexts.

File: docs/protocol/src/protocol/addresses_keys/addresses.md [hunk 2]
```diff
@@ -10,17 +10,17 @@ scanning service.
...
 
-Addresses are parameterized by *diversifiers*, 11-byte tags used to derive up to
-$2^{88}$ distinct addresses for each spending authority.  The diversifier is
+Addresses are parameterized by *diversifiers*, 16-byte tags used to derive up to
+$2^{128}$ distinct addresses for each spending authority.  The diversifier is
 included in the address, so it should be uniformly random.  To ensure this,
-diversifiers are indexed by a *address index* $i \in \{0, \ldots, 2^{88} -
+diversifiers are indexed by a *address index* $i \in \{0, \ldots, 2^{128} -
 1\}$; the $i$-th diversifier $d_i$ is the encryption of $i$ using [AES-FF1] with
 the diversifier key $\mathsf{dk}$.[^1]
```
Comment (hdevalence) [inline docs/protocol/src/protocol/addresses_keys/addresses.md:18]: We should update the AES-FF1 reference here (it's just raw AES-128 now right?)

File: testnets/026-hermippe/allocations.csv [hunk 3]
```diff
@@ -0,0 +1,4 @@
+"amount","denom","address"
+"1_000__000_000","upenumbra","penumbrav2t1fcy6crf6u4r450k8y4nye43puxet2ytfh7s0dzxsxjk68czej9mp37xv49np0clv4dc8cwg4re0xfs79uwlfehnja4p0revmlek0drezxfse8spg3qc6gux6vyuzuuls6v6mmr"
+"100","gm","penumbrav2t1fcy6crf6u4r450k8y4nye43puxet2ytfh7s0dzxsxjk68czej9mp37xv49np0clv4dc8cwg4re0xfs79uwlfehnja4p0revmlek0drezxfse8spg3qc6gux6vyuzuuls6v6mmr"
+"1","cube","penumbrav2t1fcy6crf6u4r450k8y4nye43puxet2ytfh7s0dzxsxjk68czej9mp37xv49np0clv4dc8cwg4re0xfs79uwlfehnja4p0revmlek0drezxfse8spg3qc6gux6vyuzuuls6v6mmr"
```
Comment (hdevalence) [inline testnets/026-hermippe/allocations.csv:4]: Where'd this address come from?  Is this the address for the hardcoded account used for integration tests? If not, can we add that one in?

Comment (hdevalence) [reply testnets/026-hermippe/allocations.csv:4]: NVM, resolved in next commit in sequence

File: docs/protocol/src/protocol/addresses_keys/addresses.md (hunk 2 above)
Comment (hdevalence) [reply docs/protocol/src/protocol/addresses_keys/addresses.md:18]: NVM, resolved in next commit in sequence

**Review by hdevalence — APPROVED** [review]
Very nice!!!!!

This looks great.

**Pull request merged**